### PR TITLE
Disable alerts when restarting network or node

### PR DESCRIPTION
**Pull request by fkondej**
closes vegaprotocol/jenkins-shared-library#380

### CODE REVIEW

File: vars/alert.groovy [hunk 1]
```diff
@@ -0,0 +1,99 @@
+import groovy.json.JsonSlurperClassic
+import groovy.json.JsonBuilder
+
+String disableAlerts(Map args=[:]) {
```
Comment (daniel1302) [inline vars/alert.groovy:4]: We could add a bit of documentation what are args, and maybe example call of that function

File: vars/pipelineVegavisorManageNode.groovy [hunk 2]
```diff
@@ -284,6 +296,22 @@ void call() {
...
+                        }
+                    }
+                    unsuccessful {
+                        catchError {
+                            script {
+                                alert.enableAlerts(silenceID: ALERT_SILENCE_ID, delay: 0)
+                            }
+                        }
+                    }
+                }
```
Comment (daniel1302) [inline vars/pipelineVegavisorManageNode.groovy:314]: we could use the `always` condition for that: 
https://www.jenkins.io/doc/book/pipeline/syntax/

File: vars/alert.groovy [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+
+void enableAlerts(Map args=[:]) {
+    assert args?.silenceID : "enableAlerts error: missing silenceID argument. Arguments: ${args}"
+    int delay = (args?.delay ?: "5") as int // in minutes
+
+    def now = new Date()
+    def end = new Date(now.getTime() + (args.delay * 60 * 1000))
+    String strEnd = end.format("yyyy-MM-dd'T'HH:mm:ss'Z'", TimeZone.getTimeZone('UTC'))
+
+    silenceConfig = getDisabledAlerts(silenceID: args.silenceID)
```
Comment (daniel1302) [inline vars/alert.groovy:67]: worth to check if not empty object returned

File: vars/alert.groovy [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+                    "comment": "Created by Jenkins pipeline: ${env.RUN_DISPLAY_URL}",
+                    "id": null
+                }'
+        """
+    ).trim()
+
+    print("disableAlerts response: ${strResponse}")
+
+    def response = new groovy.json.JsonSlurperClassic().parseText(strResponse)
+    def silenceID = response["silenceID"]
```
Comment (daniel1302) [inline vars/alert.groovy:52]: what if response does not have silenceID or it is nul?

**Review by daniel1302 — APPROVED** [review]

File: vars/pipelineVegavisorManageNode.groovy (hunk 2 above)
Comment (fkondej) [reply vars/pipelineVegavisorManageNode.groovy:314]: There is a different delay for each option.

**Pull request merged**